src/movies.js

```diff
@@ -1,16 +1,95 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors (movies){
+    let directors = movies.map(function(movie){
+        return movies.directors;
+    });
+    return directors;
+}   
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
 
+function howManyMovies(movies){
+    let dramaMovies = movies.filter(function(movie){
+        movie.director == "Steven Spielberg" && movie.genre.includes("Drama");
+    });
+    return dramaMovies.length;
+}
+
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
 
+function ratesAverage(movie){
+    if (movies.length == 0){
+        return 0;
+    }
+    let totalRates = movies.reduce(function(acc, movie){
+        return acc + movie.rate || 0            ;
+    }, 0);
+    return Number((totalRates/movies.length).toFixed(2));
+}
+
 // Iteration 4: Drama movies - Get the average of Drama Movies
 
+let dramaMovies = movies.filter((movie) => {
```

Comment:
This is what is breaking Jasmine. Remember that for these labs, all your code should be inside functions or objects. But declaring variables outside and doing returns it is breaking the testing system.